--- /usr/portage/mail-client/evolution/evolution-3.4.4.ebuild	2012-12-24 06:01:18.000000000 +0100
+++ /usr/local/portage/mail-client/evolution/evolution-3.4.4.ebuild	2013-01-23 16:41:06.004003079 +0100
@@ -33,6 +33,7 @@
 	>=gnome-base/gnome-desktop-2.91.3:3
 	>=gnome-base/gsettings-desktop-schemas-2.91.92
 	>=dev-libs/libgweather-2.90.0:2
+	<dev-libs/libgweather-3.5:2
 	>=media-libs/libcanberra-0.25[gtk3]
 	>=x11-libs/libnotify-0.7
 	>=gnome-extra/evolution-data-server-${PV}[gnome-online-accounts?,weather]
