From 614ae53d0090c12fc7a078639225c767ad7e7afd Mon Sep 17 00:00:00 2001
From: nobody <nobody@GentooLaptopMaxdata.(none)>
Date: Sun, 3 Jun 2012 04:43:36 +0200
Subject: Added chromium, epiphany and thunar

---
 util/fluxbox-generate_menu.in |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/util/fluxbox-generate_menu.in b/util/fluxbox-generate_menu.in
index 63c152e..c8eb53c 100755
--- a/util/fluxbox-generate_menu.in
+++ b/util/fluxbox-generate_menu.in
@@ -1522,7 +1522,7 @@ if find_it_options $MY_BROWSER; then
 else
     [ -n "$MY_BROWSER" ] && echo "Warning: you chose an invalid browser." >&2
     #The precise order is up for debate.
-    for browser in firefox mozilla-firefox mozilla-firebird MozillaFirebird opera skipstone mozilla seamonkey galeon konqueror dillo netscape w3m amaya links lynx; do
+    for browser in firefox mozilla-firefox mozilla-firebird MozillaFirebird opera skipstone mozilla chromium epiphany seamonkey galeon konqueror dillo netscape w3m amaya links lynx; do
         if find_it_options $browser; then
             DEFAULT_BROWSER=$browser
             break
@@ -1581,7 +1581,7 @@ append "[exec] (${DEFAULT_TERMNAME}) {${DEFAULT_TERM}}"
 
 case "$DEFAULT_BROWSERNAME" in
     links|w3m|lynx)  append "[exec] (${DEFAULT_BROWSERNAME}) {${DEFAULT_TERM} -e ${DEFAULT_BROWSER} ${HOMEPAGE}}" ;;
-    firefox|firebird|mozilla|seamonkey|phoenix|galeon|dillo|netscape|amaya) append "[exec] (${DEFAULT_BROWSERNAME}) {${DEFAULT_BROWSER}}" ;;
+    chromium|epiphany|firefox|firebird|mozilla|seamonkey|phoenix|galeon|dillo|netscape|amaya) append "[exec] (${DEFAULT_BROWSERNAME}) {${DEFAULT_BROWSER}}" ;;
     konqueror) append "[exec] (konqueror) {kfmclient openProfile webbrowsing}" ;;
     opera) append "[exec] (opera) {env QT_XFT=true opera}" ;;
     MozillaFirebird) append "[exec] (firebird) {MozillaFirebird}" ;;
@@ -1600,7 +1600,7 @@ append_menu_end
 
 append_submenu "${NETMENU}"
     append_submenu "${BROWSERMENU}"
-        normal_find firefox mozilla-firefox MozillaFirefox galeon mozilla seamonkey dillo netscape vncviewer
+        normal_find firefox mozilla-firefox MozillaFirefox galeon mozilla chromium epiphany seamonkey dillo netscape vncviewer
         find_it links       append "[exec]   (links-graphic) {links -driver x ${HOMEPAGE}}"
         find_it opera       append "[exec]   (opera) {env QT_XFT=true opera}"
         find_it konqueror   append "[exec]   (konqueror) {kfmclient openProfile webbrowsing}"
@@ -1676,7 +1676,7 @@ append_menu_end
 
 append_submenu "${FILEUTILSMENU}"
     find_it     konqueror append "[exec] (konqueror) {kfmclient openProfile filemanagement}"
-    normal_find gentoo krusader kcommander linuxcmd rox tuxcmd krename xfe xplore worker endeavour2 evidence
+    normal_find gentoo thunar krusader kcommander linuxcmd rox tuxcmd krename xfe xplore worker endeavour2 evidence
     find_it     nautilus append "[exec] (nautilus) {nautilus --no-desktop --browser}"
     cli_find mc
 append_menu_end
-- 
1.7.3.4

